app/src/rooms/api/serializers.py

```diff
@@ -93,13 +102,16 @@ def update(self, instance, validated_data):
 
     @staticmethod
     def _validate_beds(beds_data, available_beds_data):
-        if available_beds_data.get("single") > beds_data.get("single"):
+        if available_beds_data.get("single") > beds_data.get("single") + beds_data.get("double"):
             raise serializers.ValidationError(
-                "Cannot set more available single beds than real single beds")
+                "Cannot set more available single beds than real single beds plus double beds")
 
-        if available_beds_data.get("double") > beds_data.get("double"):
+        if available_beds_data.get("double") > \
+                beds_data.get("double") - max(0, available_beds_data.get("single")
+                                                 - beds_data.get("single")):
```

Comment:
I would suggest moving this part: `beds_data.get("double") - max(0, available_beds_data.get("single")  - beds_data.get("single"))` before `if` like this:
```python
foo = count_value
if  available_beds_data.get("double") > foo:
    sth
```